unity-renderer/Assets/DCLPlugins/ECS7/Systems/InputSenderSystem/ECSInputSenderSystem.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using DCL.ECS7;
+using DCL.ECS7.InternalComponents;
+using DCL.ECSComponents;
+using DCL.ECSRuntime;
+
+namespace ECSSystems.InputSenderSystem
+{
+    public static class ECSInputSenderSystem
+    {
+        private class State
+        {
+            public IInternalECSComponent<InternalInputEventResults> inputResultComponent;
+            public IECSComponentWriter componentWriter;
+        }
+
+        public static Action CreateSystem(
+            IInternalECSComponent<InternalInputEventResults> inputResultComponent,
+            IECSComponentWriter componentWriter)
+        {
+            var state = new State()
+            {
+                inputResultComponent = inputResultComponent,
+                componentWriter = componentWriter
+            };
+            return () => Update(state);
+        }
+
+        private static void Update(State state)
+        {
+            var inputResults = state.inputResultComponent.GetForAll();
+            var writer = state.componentWriter;
+            
+            for (int i = 0; i < inputResults.Count; i++)
+            {
+                var model = inputResults[i].value.model;
+                if (!model.dirty)
+                    continue;
+
+                var scene = inputResults[i].value.scene;
+                var entity = inputResults[i].value.entity;
+
+                PBPointerEventsResult result = new PBPointerEventsResult();
+                result.Commands.Capacity = model.events.Count;
+
+                // using foreach to iterate through queue without removing it elements
```

Comment:
why don't we remove its elements?

Reply:
because we want to keep a "history" of the latest inputs (up to 30)
from https://github.com/decentraland/sdk/issues/352

> The proposed solution unifies PointerDown and PointerUp components into one single PointerEvents component with a list of all the pointer commands executed during the time window. This approach is inspired in the [netcode presentation of Overwatch ECS](https://www.gdcvault.com/play/1024001/-Overwatch-Gameplay-Architecture-and). The list of last N commands enables extracting all the needed information to not loose any data and to enable extensibility.